Modules/_testcapimodule.c

```diff
@@ -1018,6 +1018,42 @@ test_buildvalue_N(PyObject *self, PyObject *Py_UNUSED(ignored))
 }
 
 
+static PyObject *
+test_get_statictype_slots(PyObject *self, PyObject *Py_UNUSED(ignored))
+{
+    char *tp_name = PyType_GetSlot(&PyLong_Type, Py_tp_name);
+    assert(strcmp(tp_name, "int") == 0);
+
+    newfunc tp_new = PyType_GetSlot(&PyLong_Type, Py_tp_new);
+    PyObject *args = PyTuple_New(0);
+    PyObject *object = tp_new(&PyLong_Type, args, NULL);
+    assert(object);
+
+    reprfunc tp_repr = PyType_GetSlot(&PyLong_Type, Py_tp_repr);
+    PyObject *decimal_str = tp_repr(object);
+    assert(decimal_str);
+
+    ternaryfunc tp_call = PyType_GetSlot(&PyLong_Type, Py_tp_call);
+    assert(tp_call == 0);
+
+    binaryfunc nb_add = PyType_GetSlot(&PyLong_Type, Py_tp_nb_add); 
+    PyObject *result = nb_add(object, object);
```

Comment:
You can rather assert somethiong like `nb_add == PyLong_Type->tp_as_number->nb_add`.

Reply:
no problem.